### PR TITLE
fix: checking if symlink with same prefix points outside the directory

### PR DESCRIPTION
**Pull request by mmaietta**
> If a directory is pointed to by a symbolic link whose name is the prefix of the directory, the directory won't be included in the asar. For example, the directory `aa` won't be included.
> 
> ```
> ├── a -> aa
> └── aa
> ```

Added a check (and test case) to see if the symlink is actually to a different directory and not a file (e.g. `startsWith`) within the symlinked dir. New test case mirrors `test/input/packthis-with-symlink` but with the symlink having the same prefix of a directory outside the symlink

Fixes: https://github.com/electron/asar/issues/303

### CODE REVIEW

**Review by BlackHole1 — APPROVED** [review]
LGTM

**Comment by mmaietta** [comment]
Thanks @BlackHole1 ! Would you mind also taking a look at this PR? It's the same style of fix but for asar unpacking: https://github.com/electron/asar/pull/333

**Comment by BlackHole1** [comment]
@mmaietta Already approved! Thank you for your fixed!
In general, we do not merge PRs immediately (unless they are very simple or submitted by a bot for dependency updates). We usually wait a few days to allow other maintainers to provide their input.

**Pull request merged**
